plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src-gen/org/wso2/developerstudio/eclipse/gmf/esb/parts/forms/DependencyProvider.java

```diff
@@ -1166,39 +1174,71 @@ private void showMessage(String msg) {
 
     }
 
+    public static MavenExecutionResult runMavenExecution(IFile pomFile, List<String> goals, IProgressMonitor monitor)
+            throws Exception {
+
+        if (monitor == null) {
+            monitor = new NullProgressMonitor();
+        }
+        MavenExecutionResult result = null;
+        try {
+            MavenPluginActivator.getDefault().setDebugging(true);
+            monitor.beginTask("Building: " + pomFile.getName(), 3);
+            MavenProjectManager projectManager = MavenPluginActivator.getDefault().getMavenProjectManager();
+            IMaven maven = MavenPlugin.getDefault().getMaven();
+
+            IMavenProjectFacade facade = projectManager.create(pomFile, true, new SubProgressMonitor(monitor, 1));
+            ResolverConfiguration config = facade.getResolverConfiguration();
+            MavenExecutionRequest request = projectManager.createExecutionRequest(pomFile, config,
+                    new SubProgressMonitor(monitor, 1));
+            request.setGoals(goals);
+            request.setShowErrors(true);
+            request.setUpdateSnapshots(true);
+            result = maven.execute(request, new SubProgressMonitor(monitor, 1));
+
+            return result;
+
+        } catch (Throwable e) {
+            e.printStackTrace();
```

Comment:
Use a logger and log the message along with the trace.

Reply:
fixed